homeassistant/components/home_connect/coordinator.py

```diff
@@ -153,14 +154,22 @@ def start_event_listener(self) -> None:
             f"home_connect-events_listener_task-{self.config_entry.entry_id}",
         )
 
-    async def _event_listener(self) -> None:
+    async def _event_listener(self) -> None:  # noqa: C901
         """Match event with listener for event type."""
         retry_time = 10
         while True:
             try:
                 async for event_message in self.client.stream_all_events():
                     retry_time = 10
                     event_message_ha_id = event_message.ha_id
+                    if (
+                        event_message_ha_id in self.data
+                        and not self.data[event_message_ha_id].info.connected
+                    ):
+                        self.data[event_message_ha_id].info.connected = True
+                        self._call_all_event_listeners_for_appliance(
+                            event_message_ha_id
+                        )
```

Comment:
I understand that we do this because we set all the appliances' connected attribute to `False` here:
https://github.com/home-assistant/core/blob/f2fd334615ee0d2a156098e0219228910c23a340/homeassistant/components/home_connect/coordinator.py#L265-L276
Right?
What if an appliance doesn't not send any event when stream is opened? If `EventStreamInterruptedError`, `HomeConnectRequestError` are raised, the appliance will not be shown as connected until an event from that appliance is received.

This is the case of my dishwasher; While it isn't running any program or no one does nothing on the dishwasher (physically, through the app or through the API), it doesn't send any event, it can stay without sending any event forever.

Reply:
A successful coordinator refresh will make the appliance available. So after that there's doesn't need to be an event to make the appliance connected and entities available. But if the appliance went to a non connected state for some reason, I think it's reasonable to require at least one event or successful refesh before marking it as connected again.

What do you think?

Reply:
What we should do, I think, is to forward the keep alive events too.

Reply:
We can remove the change of connected state when the event stream is interrupted non breaking, and discuss that more. It's not required to fix the reported bug.

Reply:
Do we perform a coordinator refresh after `EventStreamInterruptedError`, `HomeConnectRequestError`?
Before #139740 I know we did it, but right now we aren't AFAIK.

About forward the keep alive events, because KEEP-ALIVE events' haId is always the same than the last one sent or empty string, they can fix the issue for a particular case, but also raise another issues.
- For an user that only has one appliance, it will set the appliance as connected as soon as a KEEP-ALIVE event is received
- If an DISCONNECTED event is received from an appliance, the next KEEP-ALIVE will set the appliance as connected

Reply:
Right.

There's no refresh now after interrupted event stream.

Should we remove the change of connected state on interrupted event stream then?

Reply:
Maybe. I mean, if we remove the change, the worst that can happen is that appliances that have been connected or disconnected doesn't update their state because the stream is down, but we can solve that when we reimplement the refresh, right?

Reply:
Ok, I removed it for now.